x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/SecurityRestFilter.java

```diff
@@ -60,9 +61,14 @@ public RestHandler getConcreteRestHandler() {
 
     @Override
     public void handleRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
+        // requests with the OPTIONS method should be handled elsewhere, and not by calling {@code RestHandler#handleRequest}
+        // authn is bypassed for HTTP requests with the OPTIONS method, so this sanity check prevents dispatching unauthenticated requests
         if (request.method() == Method.OPTIONS) {
-            // CORS - allow for preflight unauthenticated OPTIONS request
-            restHandler.handleRequest(request, channel, client);
+            handleException(
+                request,
+                channel,
+                new ElasticsearchSecurityException("Cannot dispatch OPTIONS request, as they are not authenticated")
+            );
```

Comment:
This is the second most interesting part.
Because OPTIONS requests bypass authentication, this is a sanity check that unauthenticated OPTIONS requests are not dispatched.